tests/chainer_tests/optimizer_hooks_tests/test_lasso.py

```diff
@@ -3,51 +3,87 @@
 import numpy as np
 
 import chainer
-from chainer import backend
-import chainer.initializers as I
 from chainer import optimizer_hooks
 from chainer import optimizers
 from chainer import testing
-from chainer.testing import attr
+
+
+_backend_params = [
+    # NumPy
+    {},
+    {'use_ideep': 'always'},
+    # CuPy
+    {'use_cuda': True, 'cuda_device': 0},
+    {'use_cuda': True, 'cuda_device': 1},
+    # ChainerX
+    {'use_chainerx': True, 'chainerx_device': 'native:0'},
+    {'use_chainerx': True, 'chainerx_device': 'cuda:0'},
+    {'use_chainerx': True, 'chainerx_device': 'cuda:1'},
+]
 
 
 class SimpleLink(chainer.Link):
 
-    def __init__(self, w, g):
+    def __init__(self, params):
         super(SimpleLink, self).__init__()
         with self.init_scope():
-            self.param = chainer.Parameter(I.Zero(), w.shape)
-        self.param.data = w
-        self.param.grad = g
+            for i, p in enumerate(params):
+                setattr(self, 'p{}'.format(i), p)
 
 
+@testing.backend.inject_backend_tests(None, _backend_params)
+@testing.backend.inject_backend_tests(None, _backend_params)
+@testing.backend.inject_backend_tests(None, _backend_params)
 class TestLasso(unittest.TestCase):
     def setUp(self):
-        self.target = SimpleLink(
-            np.arange(6, dtype=np.float32).reshape(2, 3),
-            np.arange(3, -3, -1, dtype=np.float32).reshape(2, 3))
-
-    def check_lasso(self):
-        w = self.target.param.data
-        g = self.target.param.grad
-        xp = backend.get_array_module(w)
+        num_params = 3
+        arrs = [
+            np.random.uniform(-3, 3, (2, 3)).astype(np.float32)
+            for _ in range(num_params)]
+        grads = [
+            np.random.uniform(-3, 3, (2, 3)).astype(np.float32)
+            for _ in range(num_params)]
+        params = []
+        for arr, grad in zip(arrs, grads):
+            param = chainer.Parameter(arr)
+            param.grad = grad
+            params.append(param)
+
+        self.target = SimpleLink(params)
+
+    def check_lasso(self, backend_configs):
+        target = self.target
+        assert len(backend_configs) == len(list(target.params()))
+        devices = [bc.device for bc in backend_configs]
+
         decay = 0.2
-        expect = w - g - decay * xp.sign(w)
 
+        expects = []
+        # Compute expected
+        for param, device in zip(target.params(), devices):
+            expects.append(param.array - param.grad -
+                           decay * np.sign(param.array))
```

Comment:
From the error, I guess this should be `device.xp` instead of `np`? @emcastillo does that seem correct?

Reply:
param.arrays are initially numpy arrays, with device.xp probably it will fail.
I tested it locally.

Reply:
The error is due to ```  @testing.attr.gpu
    def test_lasso_gpu(self, backend_config0,
                       backend_config1, backend_config2):
        self.target.to_gpu()```

once you remove the test_lasso_gpu as explained below it should work fine.

Reply:
Oh. Thanks. 
Any clue as to why the tests ran successfully on my system? I would expect them to fail there as well.

Reply:
python2 issues, only failed with python2, probably some cupy or numpy internal issue

Reply:
Oh. I see. I tested on Python 3. Thank You very much